Assets/MediaSoup/Transport.cs

```diff
@@ -247,17 +261,27 @@ public async Task UpdateIceServers(List<RTCIceServer> iceServers)
                 throw new ArgumentNullException("missing iceParameters");
             }
 
-            _ = awaitQueue.Push<bool>(async () =>
-            {
-                //await handlerInterface.UpdateIceServers(iceServers);
-                return true;
+            await awaitQueue.Push<bool>(AddUpdateIceServersInQueue,null,iceServers);
 
-            }, "transport.UpdateIceServers");
+        }
 
+        private async Task<bool> AddUpdateIceServersInQueue(params object[] args) 
+        {
+            List<RTCIceServer> iceServers = args[0] as List<RTCIceServer>;
+            await handlerInterface.UpdateIceServers(iceServers);
+            return true;
         }
 
-        public async Task<Producer<ProducerAppData>> ProduceAsync<ProducerAppData>(Func<Unity.WebRTC.TrackKind, RtpParameters, AppData, Task<string>> GetProducerIdCallback,
-                                    ProducerOptions<ProducerAppData> options = null) where ProducerAppData : AppData, new()
+
+        /// <summary>
+        /// This function will add the Consumer creation process in a queue and return the result in callback
+        /// </summary>
+        /// <param name="GetProducerIdCallback">Process of getting DataProducer id from server</param>
+        /// <param name="producerCallback">callback with result</param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public async Task ProduceAsync(Func<Unity.WebRTC.TrackKind, RtpParameters, AppData, Task<string>> GetProducerIdCallback,

```

Comment:
Nice to add these summaries, will help out other developers building upon this 👍